iree/turbine/kernel/wave/global_to_shared_gathers.py

```diff
@@ -172,18 +172,32 @@ def construct_shared_read_mapping(read: Read) -> IndexMapping:
     return new_mapping
 
 
-def update_read_mapping_dynamic_values(read: Read) -> list[fx.Node]:
+def update_read_mapping_dynamic_values(read: Read):
     """
     Since every thread is now only reading one element along the gather dimension,
     we need to modify the read of the dynamic value to read only one element
     and do it at the specified index.
     """
+    new_dyn_vals = []
     for dyn_val in read.mapping_dynamic_vals:
         custom = get_custom(dyn_val)
-        custom.update_arg("elements_per_thread", 1)
-        for dim in dyn_val.index:
-            if dim in read.index:
-                dyn_val.index[dim] = read.index[dim]
+        assert isinstance(custom, Read), f"Only read nodes are supported, got {custom}"
+        # We cannot just update dynamic val inplace, as it may be used by other nodes
```

Comment:
Can you move this comment to the function comment? Also can you add some comments saying that we generate redundant reads but expect them to be CSE-ed downstream ?